core/src/main/java/com/radixdlt/consensus/vertexstore/VertexStoreJavaImpl.java

```diff
@@ -302,91 +335,97 @@ public InsertVertexChainResult insertVertexChain(VertexChain vertexChain) {
         }
       }
 
-      insertVertex(v).onPresent(bftInsertUpdates::add);
+      final var insertRes = insertVertexInternal(v);
+      if (insertRes.isSuccess()) {
+        bftInsertUpdates.add(insertRes.unwrap());
+      } else {
+        switch (insertRes.unwrapError()) {
+          case ALREADY_PRESENT, PREPARE_FAILED -> {
+            // No-op, continue iterating the vertices
```

Comment:
Why is `PREPARE_FAILED` a no-op rather than a fatal error? Do we expect prepare to ever fail here?

* Is this vertex chain untrusted, so preparation failures are expected? In which case, it seems weird we handle this by continuing to iterate the vertices?
* Or has quorum signed this? In which case, this would be a divergence / (possible) byzantine quorum exception?